ravenframework/Optimizers/SimulatedAnnealing.py

```diff
@@ -248,9 +250,28 @@ def initialize(self, externalSeeding=None, solutionExport=None):
     for var in self.toBeSampled:
       self.info['amp_'+var] = None
       self.info['delta_'+var] = None
-    # queue up the first run for each trajectory
+
+    initialValues = self._initialValues
     for traj, init in enumerate(self._initialValues):
-      self._submitRun(init,traj,self.getIteration(traj))
+      values = {}
+      init = self.denormalizeData(init)
+      initialValues[traj] = self.denormalizeData(initialValues[traj])
+      for var in init:
+        if var in self.toBeSampled and self.distDict[var].distType == distType.discrete:
+          val = init[var]
+          values[var] = self.distDict[var].ppf(self.distDict[var].cdf(val))
+          if not  np.isclose(initialValues[traj][var], values[var], 1e-4):
+            self.raiseAWarning(f"Traj: {traj}. Variable {var} is associated with a discrete distribution. The inputted initial value {initialValues[traj][var]} "
+                               f"is not among available discrete values. Closest value is {values[var]}")
+        else:
+          values[var] =  init[var]
+      values = self.normalizeData(values)
+
+      self._submitRun(values,traj,self.getIteration(traj))
```

Comment:
It seems to me these lines can be moved to Optimizer base class.

Reply:
I would open an issue to track this in case the RAVEN team is willing to go in that direction